File: pomopetsLanding/src/types/pomodoro.ts
export interface StudySession {
  id: string
  user_id: string
  title: string
  status: 'active' | 'finished'
  work_duration: number    // minutes
  break_duration: number   // minutes
  coins_earned: number
  rounds_completed: number
  total_work_seconds: number
  started_at: string
  finished_at: string | null
}

export interface StudyTask {
  id: string
  session_id: string
  user_id: string
  text: string
  completed: boolean
  created_at: string
}

export interface PomodoroRound {
  id: string
  session_id: string
  user_id: string
  round_number: number
  phase: 'work' | 'break'
  duration_seconds: number
  actual_seconds: number
  completed: boolean
  started_at: string
  finished_at: string | null
}

export interface TimerSnapshot {
  sessionId: string
  phase: 'work' | 'break'
  roundNumber: number
  remainingSeconds: number
  timerStatus: 'running' | 'paused'
  totalWorkSeconds: number
  coinsAwardedMarks: number
  workDurationSec: number
  breakDurationSec: number
  pauseStartedAt: number | null
  savedAt: number
}

File: ios/PomoPets/PomoPets/Models/Pomodoro.swift
import Foundation

struct StudySession: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let title: String
    var status: String
    let workDuration: Int
    let breakDuration: Int
    var coinsEarned: Int
    var roundsCompleted: Int
    var totalWorkSeconds: Int
    let startedAt: String
    var finishedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, status
        case userId = "user_id"
        case workDuration = "work_duration"
        case breakDuration = "break_duration"
        case coinsEarned = "coins_earned"
        case roundsCompleted = "rounds_completed"
        case totalWorkSeconds = "total_work_seconds"
        case startedAt = "started_at"
        case finishedAt = "finished_at"
    }
}

struct StudyTask: Codable, Identifiable, Sendable {
    let id: String
    let sessionId: String
    let userId: String
    let text: String
    var completed: Bool
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, text, completed
        case sessionId = "session_id"
        case userId = "user_id"
        case createdAt = "created_at"
    }
}
